cmd/soroban-rpc/internal/ledgerentry_storage/ledgerentry_storage.go

```diff
@@ -171,19 +191,35 @@ func (ls *ledgerEntryStorage) run(ctx context.Context, archive historyarchive.Ar
 	nextLedger := startLedger + 1
 	for {
 		ls.logger.Infof("Applying txmeta ledger entries changes for ledger %d", nextLedger)
-		reader, err := ingest.NewLedgerChangeReader(ctx, ledgerBackend, ls.networkPassPhrase, nextLedger)
+		ledger, err := ledgerBackend.GetLedger(ctx, nextLedger)
 		if err != nil {
 			if errors.Is(err, context.Canceled) {
 				// we were told to stop
 				return
 			}
 			panic(err)
 		}
+
+		reader, err := ingest.NewLedgerChangeReaderFromLedgerCloseMeta(ls.networkPassPhrase, ledger)
+		if err != nil {
+			panic(err)
+		}
+		if err := ls.events.IngestEvents(reader.LedgerTransactionReader); err != nil {
+			panic(err)
+		}
+
 		tx, err := ls.db.NewLedgerEntryUpdaterTx(nextLedger, maxBatchSize)
 		if err != nil {
 			panic(err)
 		}
+		if err = tx.TrimLedgers(ls.retentionWindow); err != nil {
+			panic(err)
+		}
+		if err = tx.InsertLedger(ledger); err != nil {
+			panic(err)
+		}
 
+		reader.Rewind()
```

Comment:
seems wrong that we need to rewind and re-process. But maybe I'm not understanding the codeflow here correctly.